File: Programmers/Level3/순위/순위.js
function solution(n, results) {
    const INF = Infinity;
    let matrix = [...Array(n + 1)].map(() => [...Array(n + 1)].fill(INF));

    for (let [win, lose] of results) {
        matrix[win][lose] = true;
        matrix[lose][win] = false;
    }

    for (let k = 0; k <= n; k++) {
        for (let i = 0; i <= n; i++) {
            for (let j = 0; j <= n; j++) {
                if (matrix[i][k] === INF) continue;

                if (matrix[i][k] === matrix[k][j]) {
                    matrix[i][j] = matrix[i][k];
                    matrix[j][i] = !matrix[i][k];
                }
            }
        }
    }

    return [...matrix.map((m) => m.filter((v) => v === INF).length)].filter(
        (v) => v === 2
    ).length;
}

File: Programmers/Level3/순위/순위.py
def solution(n, results):
    INF = int(1e9)
    matrix = [[INF for _ in range(n + 1)] for _ in range(n + 1)]
    
    for win, lose in results:
        matrix[win][lose] = True
        matrix[lose][win] = False
    
    for k in range(n + 1):
        for i in range(n + 1):
            for j in range(n + 1):
                if matrix[i][k] == INF:
                    continue
                    
                if matrix[i][k] == matrix[k][j]:
                    matrix[i][j] = matrix[i][k]
                    matrix[j][i] = not matrix[i][k]
    
    return len([m for m in matrix if m.count(INF) == 2])